src/Microsoft.OData.Core/JsonLight/ReorderingJsonReader.cs

```diff
@@ -31,6 +31,38 @@ internal ReorderingJsonReader(IJsonReader innerReader, int maxInnerErrorDepth)
             Debug.Assert(innerReader != null, "innerReader != null");
         }
 
+        /// <summary>
+        /// Creates a stream for reading a stream value.
+        /// </summary>
+        /// <returns>A Stream used to read a stream value.</returns>
+        public override Stream CreateReadStream()
+        {
+            Stream result = new MemoryStream(this.Value == null ? new byte[0] :

```

Comment:
>MemoryStream [](start = 32, length = 12)

1. Do we need to cache the MemoryStream, always create a new one? Same for the TextReader?

2. How to dispose it? Or How to make sure customer dispose it or not?

Reply:
We need a new MemoryStream because it will be a new value.
Streams returned from CreateReadStream() get wrapped by an ODataNotificationStream in ODataReaderCore that ensures it gets disposed.

---
In reply to: [266669354](https://github.com/OData/odata.net/pull/1204#discussion_r266669354) [](ancestors = 266669354)